README.md

```diff
@@ -1,11 +1,48 @@
+Instead of tallying every vote publicly, ALBA allows an aggregator (like an election authority) to generate a compact proof showing that a sufficient number of valid votes has been cast for each candidate to meet the required threshold for a decision (such as reaching a quorum or winning a majority).
+Combining ALBA with zero-knowledge technology such as _zkSNARKs_, we can efficiently keep individual votes private while enabling quick, efficient validation of the voting outcome, making it particularly useful for secure, private, and scalable voting systems where privacy and efficiency are critical.
+
+The paper introduces various ALBA protocol constructions tailored to different needs.
+The basic construction allows a prover to show possession of a large set by creating a proof sequence of elements that meet staged hash-based conditions, efficiently excluding small sets.
+_Pre-hashing_ improves this by precomputing hashes to group elements into _bins_, reducing computation and enhancing performance, especially with large sets.
+For smaller sets, the generalized Telescope scheme allows multiple attempts to form a proof, adjusting parameters to maintain efficiency across sizes.
+Decentralized versions of ALBA include the _Simple Lottery Construction_, where each party holding an element decides to share it based on a random _lottery_ mechanism, with an aggregator forming the proof from a target number of shared elements.
+The _Decentralized Telescope_ adapts the Telescope scheme for multiple parties, who individually apply it and share qualifying elements with an aggregator.
+Finally, the weighted extension supports elements with integer weights, allowing the prover to meet a total weight threshold, making ALBA versatile for contexts where elements have varying significance.
+
+The library covers the following constructions of the ALBA protocol:
+1. Centralized Telescope
+2. Simple Lottery Construction
+3. Decentralized Telescope
+4. Wighted-Decentralized Telescope
+
+### Disclaimer
 
-> [!IMPORTANT]
 > This code is NOT fit for production, it's not been optimised, thoroughly tested, nor audited by competent cryptographers.
-> Its one and only purpose is to help people who are more familiar with code than equations to have a better understanding of ALBAs
+> Its one and only purpose is to help people who are more familiar with code than equations to prototype larger protocols using ALBA.
+
+### Documentation
+👉 We deliver comprehensive [documentation][crate::docs] aimed at connecting theory with practical implementation.
+
+👉 Checkout website on this [link](https://alba.cardano-scaling.org).
+
+### Compiling the library
+Compile the library:
+```shell
+cargo build --release
+```
 
-👉 Checkout documentation on https://alba.cardano-scaling.org
+### Tests and Benchmarks
+Run tests with `cargo test`. Run benchmarks with `cargo bench`. 
```

Comment:
perhaps we should add a command that compiles the documentation?
`cargo doc --no-deps --open`